ISHLogDNAService/ISHLogDNAService.m

```diff
@@ -101,6 +102,12 @@ + (instancetype)messageWithLine:(NSString *)line level:(ISHLogDNALevel)level met
         [errorDict setObject:error.localizedFailureReason forKey:ISHLogDNAServiceKeyErrorDescription];
     }
 
+    NSError *underlyingError = error.userInfo[NSUnderlyingErrorKey];
```

Comment:
how about recursion? The underlying error might have an underlying error. And so on.